### ISSUE TITLE
Some success running toolbox on Ubuntu 18.04 (first experiment)

### ISSUE THREAD

**Issue by eriksjolund** [debarshiray/toolbox#197]
I just want to mention that I had some success running `toolbox` on Ubuntu 18.04.

First log in to an Ubuntu 18.04 workstation: 

```
myuser@laptop~$ ssh -XC testuser@workstation
```

And then run `toolbox` from this branch
https://github.com/eriksjolund/toolbox/tree/ubuntu-modifications

```
testuser@workstation:~$  cat /etc/issue
Ubuntu 18.04.2 LTS \n \l

testuser@workstation:~$ toolbox create  --release f30 -c mytoolbox
Created container: mytoolbox
Enter with: toolbox enter --container mytoolbox
testuser@workstation:~$ toolbox enter --container mytoolbox
testuser@toolbox:~$ sudo dnf install -y --quiet xorg-x11-apps 

We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

testuser@toolbox:~$ sudo dnf install -y --quiet xorg-x11-apps 
testuser@toolbox:~$ xclock 
X11 connection rejected because of wrong authentication.
Error: Can't open display: localhost:15.0
testuser@toolbox:~$ sudo hostname workstation 
testuser@toolbox:~$ xclock 
```

`xclock` showed a clock in a graphical window. As you see I had to run `sudo hostname workstation` to get the X11 authentication working.   `workstation` is the hostname of the host.

Note, this was  just a quick experiment. 

The branch contains some modifications related to paths:
https://github.com/eriksjolund/toolbox/tree/ubuntu-modifications

(Most probably the modifications break normal Fedora support)

**Comment by debarshiray** [comment]
Interesting.

See https://github.com/storaged-project/udisks/pull/641 about `/run/media`. Toolbox itself carries a `tmpfiles.d` snippet to create the directory unless a newer udisks gets widely adopted.

**Comment by debarshiray** [comment]
Regarding `/run/host/etc/resolv.conf` there's https://github.com/debarshiray/toolbox/issues/187

**Comment by debarshiray** [comment]
I don't know anything about the X11 authentication problem right now.

**Comment by markstos** [comment]
@eriksjolund Did you also try to run Ubuntu containers inside of toolbox?

**Comment by eriksjolund** [comment]
@markstos No, I haven't tried to run Ubuntu containers inside of toolbox.

Has anyone had success with ssh -X?

Two weeks ago I made the same test from a Fedora 30 laptop.
ssh -X testuser@fedora30workstation

If I remember correctly I saw the same X11 authentication problem happening there too.

**Comment by markstos** [comment]
You might also try the `tlbx` fork: https://gitlab.com/uppercat/tlbx

**Comment by bam80** [comment]
I thought Toolbox is a cross-distro app, it is not apparently?
So I can't just install the main version on Ubuntu currently?

**Comment by debarshiray** [comment]
> I thought Toolbox is a cross-distro app, it is not apparently?

One problem is that Toolbox has a client-side `/usr/bin/toolbox` binary and OCI images that are maintained on some registry. Somebody needs to take ownership of the Ubuntu OCI image on an ongoing long-term basis. Part of this is *upstreaming* bugs to Ubuntu (eg., ensuring that `/etc/localtime` is a [relative](https://github.com/containers/toolbox/issues/622) symbolic link), and digging into Ubuntu specific issues.

Right now, neither me nor @HarryMichal has the bandwidth to take on this extra responsibility. So somebody else needs to step up.

Also, it's not exactly an application. It's some weird hybrid between a container tool and an interactive shell, that interacts quite deeply with some of the innards of the host operating system.

**Comment by debarshiray** [comment]
Closing.

Earlier this year @eriksjolund told me via email that he is too busy to maintain the Ubuntu support long-term. We can't merge things that have nobody maintaining them.

**Comment by debarshiray** [comment]
Let's focus on https://github.com/containers/toolbox/pull/483 which is more active. :)

**Comment by bam80** [comment]
> One problem is that Toolbox has a client-side /usr/bin/toolbox binary and OCI images that are maintained on some registry. Somebody needs to take ownership of the Ubuntu OCI image on an ongoing long-term basis. Part of this is upstreaming bugs to Ubuntu (eg., ensuring that /etc/localtime is a relative symbolic link), and digging into Ubuntu specific issues.

> Right now, neither me nor @HarryMichal has the bandwidth to take on this extra responsibility. So somebody else needs to step up.

Isn't it what https://github.com/89luca89/distrobox is about?

> It implements the same concepts introduced by https://github.com/containers/toolbox but in a simplified way using POSIX sh and aiming at broader compatibility.

It also seem has galore of host/guest distros support:
https://github.com/89luca89/distrobox#host-distros

So the question is do we really need to maintain two projects in parallel doing mostly exactly the same things?
